core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java

```diff
@@ -320,31 +393,20 @@ public List<ActiveCompaction> getActiveCompactions(String server)
   public List<ActiveCompaction> getActiveCompactions()
       throws AccumuloException, AccumuloSecurityException {
 
-    Map<String,Set<HostAndPort>> compactors = ExternalCompactionUtil.getCompactorAddrs(context);
-    List<String> tservers = getTabletServers();
+    Set<ServerId> compactionServers = new HashSet<>();
+    compactionServers.addAll(getServers(ServerTypeName.COMPACTOR));
+    compactionServers.addAll(getServers(ServerTypeName.TABLET_SERVER));
 
-    int numThreads = Math.max(4, Math.min((tservers.size() + compactors.size()) / 10, 256));
+    int numThreads = Math.max(4, Math.min((compactionServers.size()) / 10, 256));
```

Comment:
Java 21 finally gets a nice way of expressing this common pattern of returning a value bounded on both ends:

https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Math.html#clamp(long,int,int)